tutorials/cassandra/src/main/java/com/datatorrent/cassandra/CassandraInputOperator.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2016 DataTorrent, Inc.
+ * All rights reserved.
+ */
+package com.datatorrent.cassandra;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import com.datatorrent.api.Context.OperatorContext;
+import com.datatorrent.api.DefaultInputPort;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.Operator.IdleTimeHandler;
+import com.datatorrent.api.annotation.InputPortFieldAnnotation;
+import com.datatorrent.contrib.cassandra.CassandraPOJOInputOperator;
+import com.datatorrent.contrib.cassandra.CassandraTransactionalStore;
+
+/**
+ * 
+ * Extends {@link CassandraPOJOInputOperator} Reads data from Cassandra DB.<br/>
+ * Operator Ports: <br/>
+ * 1. scanTrigger: Receives {@link ScanInformation} from upstream DataPopulator.
+ * The operator waits to read data till the trigger is received. Upstream data
+ * populator sends the trigger when it is done writing data to database.<br/>
+ * 2. validatationInfo: forwards {@link ScanInformation} received on scanTrigger
+ * port to downstream operator.
+ */
+public class CassandraInputOperator extends CassandraPOJOInputOperator implements IdleTimeHandler
+{
+  private static Logger LOG = LoggerFactory.getLogger(CassandraInputOperator.class);
+  private int tuplesCount;
+  private static int processedTuples;
+  private ScanInformation scanInfo = null;
+  private long scanStartTime;
+  @InputPortFieldAnnotation(optional = true)
+  public transient DefaultInputPort<ScanInformation> scanTrigger = new DefaultInputPort<ScanInformation>()
```

Comment:
I wanted this to act as trigger. Cassandra db guaranties eventual consistency, so need the reader operator to wait till writer has done it's job. Let me check if I can make producer send a single to input operator directly.